File: src/main/java/pl/training/workshop/m5/s07_collapsehierarchy/step1/Hall.java
package pl.training.workshop.m5.s07_collapsehierarchy.step1;

/** Krok 1: bez zmian. */
public class Hall {
    private final String name;
    private final int rows;
    private final int seatsPerRow;
    private final int vipFromRow;

    public Hall(String name, int rows, int seatsPerRow, int vipFromRow) {
        this.name = name;
        this.rows = rows;
        this.seatsPerRow = seatsPerRow;
        this.vipFromRow = vipFromRow;
    }

    public String name() {
        return name;
    }

    public int capacity() {
        return rows * seatsPerRow;
    }

    public boolean isVip(int row) {
        return row >= vipFromRow;
    }

    public String describe() {
        return name + ": " + capacity() + " miejsc, VIP od rzędu " + vipFromRow;
    }
}

File: csharp/src/Training.Workshop/M5/S07CollapseHierarchy/Step1/Hall.cs
namespace Training.Workshop.M5.S07CollapseHierarchy.Step1;

/// <summary>Krok 1: bez zmian.</summary>
public class Hall
{
    private readonly string _name;
    private readonly int _rows;
    private readonly int _seatsPerRow;
    private readonly int _vipFromRow;

    public Hall(string name, int rows, int seatsPerRow, int vipFromRow)
    {
        _name = name;
        _rows = rows;
        _seatsPerRow = seatsPerRow;
        _vipFromRow = vipFromRow;
    }

    public string Name => _name;

    public virtual int Capacity => _rows * _seatsPerRow;

    public bool IsVip(int row)
    {
        return row >= _vipFromRow;
    }

    public virtual string Describe()
    {
        return _name + ": " + Capacity + " miejsc, VIP od rzędu " + _vipFromRow;
    }
}
